components/List/List.js

```diff
@@ -1,16 +1,28 @@
-import { StyledList, StyledItem } from "@/styles";
+import { StyledList, StyledItem, StyledDeleteButton } from "@/styles";
+
+export default function List({ transactions, setTransactions }) {
+  function deleteTransaction(id) {
+    const updatedTransactions = transactions.filter(
+      (transaction) => transaction.id !== id
+    );
+    setTransactions(updatedTransactions);
```

Comment:
Something I've missed earlier:
Please don't pass setter functions down to components. 🙅‍♀️ Please write a handler function at the top level of your app that calls the setter functions and pass the handler function along instead.